docker-registry-login/action.yaml

```diff
@@ -0,0 +1,36 @@
+name: 'Docker registry login'
+description: 'Log in to a Docker registry, DockerHub or custom one, depending on inputs with retries'
+
+inputs:
+  registry:
+    description: 'Registry URL, if not provided defaults to DockerHub'
+    required: false
+  username:
+    description: 'Username for given registry or DockerHub'
+    requred: false
+  password:
+    description: 'Password or preferably PAT for "username"'
+    required: false
+  attempt_limit:
+    description: 'Maximum number of total attempts'
+    required: false
+    default: 3
+  attempt_delay:
+    description: 'Delay between attempts'
+    required: false
+    default: 3000
+
+runs:
+  using: 'composite'
+
+  steps:
+    - name: Log in to Docker registry
+      uses: Wandalen/wretry.action/main@8ceaefd717b7cdae4f2637f9a433242ade421a0a  # 3.7.2
```

Comment:
I'd consider using a fork with this fix https://github.com/docker/login-action/pull/803 instead of introducing a new action (which could break in new ways, here's [an example](https://github.com/Wandalen/wretry.action/issues/168))

Reply:
We've had a discussion on this with @fedordikarev with the conclusion that once that PR is merged we can remove this retry.

In that case we'll still have the benefit of containing this in one place and not having to update many-many usages of the `docker/login-action` in our own repos.

Because of this and similar issues I excluded the retries on pre and post actions, see the `/main` in this line.

Reply:
> We've had a discussion on this with @fedordikarev with the conclusion that once that PR is merged we can remove this retry.

Ah, nice! Would've been also nice to include a bit more context to the PR description

> In that case we'll still have the benefit of containing this in one place and not having to update many-many usages of the docker/login-action in our own repos.

We still can call a fork from `neondatabase/dev-actions/docker-registry-login` (I think it's a great idea to have it in a separate action 👍 )

I'm not adamant about changing the current approach, just suspicious while introducing new actions